sema4ai/src/sema4ai_code/vendored_deps/package_deps/analyzer.py

```diff
@@ -441,6 +442,56 @@ def _iter_external_endpoints_issues(self) -> Iterator[_DiagnosticsTypedDict]:
                                     "message": f"Error: 'port' must be a valid integer between 0 and 65535 if provided in a rule. Found: {rule['port']!r} (type: {type_repr(rule['port'])})",
                                 }
 
+    def _iter_version_consistency_issues(self) -> Iterator[_DiagnosticsTypedDict]:
+        """Check if the package.yaml version matches the action package version in agent-spec.yaml."""
+        import yaml
+
+        data = self._yaml_data
+        if not data:
+            return
+
+        version = data.get("version")
+        if not version:
+            return  # Version check already reported in _load_yaml_info
+
+        agent_spec_path = (
```

Comment:
I think this is not an ideal approach, rather, I think it should search for it in the parents (up to a point -- who knows how that structure will change over time).

```
check_dir = pathlib.Path(self.path).parent.parent
for i in range(6):
   agent_spec_path = check_dir / 'agent-spec.yaml
   if agent_spec_path.exists():'
     break
    if not check_dir.parent or check_dir.parent == check_dir:
        return
else:
    return
        
```